app/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -332,6 +324,94 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void configureStorageClient(AWSKinesisVideoWebRTCStorageClient storageClient) {
+        storageClient.setRegion(Region.getRegion(mRegion));
+        storageClient.setSignerRegionOverride(mRegion);
+        storageClient.setServiceNameIntern("kinesisvideo");
+        storageClient.setEndpoint(webrtcEndpoint);
+    }
+
+    private void joinStorageSession(AWSKinesisVideoWebRTCStorageClient storageClient) {
+        if (master) {
+            joinAsMaster(storageClient);
+        } else {
+            joinAsViewer(storageClient);
+        }
+    }
+
+    private void joinAsMaster(AWSKinesisVideoWebRTCStorageClient storageClient) {
+        JoinStorageSessionRequest request = new JoinStorageSessionRequest()
+                .withChannelArn(mChannelArn);
+        storageClient.joinStorageSession(request);
+        Log.i(TAG, "Master: Join storage session request sent for channel: " + mChannelArn);
+    }
+
+    private void joinAsViewer(AWSKinesisVideoWebRTCStorageClient storageClient) {
+        JoinStorageSessionAsViewerRequest request = new JoinStorageSessionAsViewerRequest()
+                .withChannelArn(mChannelArn)
+                .withClientId(mClientId);
+        storageClient.joinStorageSessionAsViewer(request);
+        Log.i(TAG, "Viewer: Join storage session request sent for channel: " + mChannelArn);
+    }
+
+    private boolean isStorageViewer() {
+        return !master && webrtcEndpoint != null;
+    }
+
+    private boolean isStorageSession() {
+        return webrtcEndpoint != null;
+    }
+    
+    private boolean checkCodecSupport() {
+        // Need to create temporary EglBase for codec checking
+        EglBase tempEglBase = EglBase.create();
+        
+        final VideoDecoderFactory vdf = new DefaultVideoDecoderFactory(tempEglBase.getEglBaseContext());
+        boolean hasH264Decoder = false;
+        for (final VideoCodecInfo videoCodecInfo : vdf.getSupportedCodecs()) {
+            if ("H264".equalsIgnoreCase(videoCodecInfo.name)) {
+                hasH264Decoder = true;
+                break;
+            }
+        }
+        
+        final VideoEncoderFactory vef = new DefaultVideoEncoderFactory(tempEglBase.getEglBaseContext(),
+                ENABLE_INTEL_VP8_ENCODER, ENABLE_H264_HIGH_PROFILE);
+        boolean hasH264Encoder = false;
+        for (final VideoCodecInfo videoCodecInfo : vef.getSupportedCodecs()) {
+            if ("H264".equalsIgnoreCase(videoCodecInfo.name)) {
```

Comment:
Can we still keep the log line where we print out all the available encoders & decoders on the device?